components/dialog/dialog-confirm.js

```diff
@@ -77,7 +77,7 @@ class DialogConfirm extends DialogMixin(LitElement) {
 						<div><h2 id="${this._titleId}" class="d2l-heading-3">${this.titleText}</h2></div>
 					</div>` : null}
 				<div id="${this._textId}" class="d2l-dialog-content">
-					<div>${this.text ? this.text.split('\n').map(line => html`<p>${line}</p>`) : null}</div>
+					<div>${this?.text?.split ? this.text.split('\n').map(line => html`<p>${line}</p>`) : this.text}</div>
```

Comment:
Previously when the check for `this.text` returned false we set the div contents to `null`. With this change we set it to `this.text` instead if there either is not `this.text` or there isn't `this.text.split`, which is potentially different. Are there any concerns around that?

Otherwise this makes sense to me.

Reply:
Agree with Margaree here that this is definitely different behaviour. Another option:

```javascript
let content = nothing;
if (typeof this.text === 'string') {
  content = this.text.split('\n').map(line => html`<p>${line}</p>`);
} else if( this.text instanceof HTMLElement ){
  content = this.text;
}
```

But I still have some concerns -- will message in Slack.

Reply:
Exactly yeah. Previously if a consumer tried to set the property to a DOM node, they would have gotten an error. With this change, they could do that, even though we don't really want them to. We'd effectively be making an exception to enable migrating our MVC inline dialog rendering while maintaining their original lang-terms.